packages/kbn-elastic-agent-utils/src/agent_guards.ts

```diff
@@ -7,32 +7,50 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { JAVA_AGENT_NAMES, OPEN_TELEMETRY_AGENT_NAMES, RUM_AGENT_NAMES } from './agent_names';
+import {
+  ANDROID_AGENT_NAMES,
+  IOS_AGENT_NAMES,
+  JAVA_AGENT_NAMES,
+  OPEN_TELEMETRY_AGENT_NAMES,
+  RUM_AGENT_NAMES,
+} from './agent_names';
 
 import type {
+  AndroidAgentName,
+  IOSAgentName,
   JavaAgentName,
   OpenTelemetryAgentName,
   RumAgentName,
   ServerlessType,
 } from './agent_names';
 
+export function hasOpenTelemetryPrefix(agentName?: string, language: string = '') {
+  if (!agentName) {
+    return false;
+  }
+
+  return (
+    agentName.startsWith(`opentelemetry/${language}`) || agentName.startsWith(`otlp/${language}`)
+  );
+}
+
 export function isOpenTelemetryAgentName(agentName: string): agentName is OpenTelemetryAgentName {
   return (
-    agentName?.startsWith('opentelemetry/') ||
+    hasOpenTelemetryPrefix(agentName) ||
```

Comment:
Does `agentName` always end with `/`?

Reply:
For OpenTelemetry agents, yes, they always have the prefix `otlp/` or `opentelemetry/`. The only cases where we could have just `otlp` or `opentelemetry` is being handled [here](https://github.com/rmyz/kibana/blob/180444-apm-ui-fix-opentelemetry-agent-names/packages/kbn-custom-icons/src/components/agent_icon/get_agent_icon.ts#L110)